### DISCUSSION TITLE
UploadFile is never None, can't `if not file:` as it's always something

### DISCUSSION THREAD

**Discussion by regisin** [fastapi/fastapi#6029] [Questions]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request

app = FastAPI()

@app.post('/up')
async def upload_post(text_file: UploadFile | None = File(default=None),
                      audio_file: UploadFile | None = None):
    print(text_file, audio_file)
    print(text_file.file, audio_file.file)
    if not text_file:
        return "Error: please provide text file."
    if not audio_file:
        return "Error: please provide audio file."
    count = char_count(text_file)
    if count > settings.CHAR_LIMIT:
        return f'Error: your text must be less than 100 characters ({count} found instead).'
    duration = audio_duration(audio_file)
    if duration < 10:
        return f'Error: could not process audio (either corrupted or too short; must be 10s or longer).'


@app.get('/up', response_class=HTMLResponse)
def upload_get() -> HTMLResponse:
    html = """
    <html>
        <body>
            <form method="post" action="/upload/up" enctype="multipart/form-data">
                <label for="text_file">Select text file: </label> <input type="file" name="text_file" accept="text/plain">
                <label for="audio_file">Select audio file: </label> <input type="file" name="audio_file" accept="audio/mpeg, audio/mp3, audio/wav">
                <input type="submit" value="Submit"/>
            </form>
        </body>
    </html>
    """
    return html
```

## Documentation issue
According to [FastAPI's docs](https://fastapi.tiangolo.com/tutorial/request-files/?h=upload#optional-file-upload), I can check if an UploadFile is submitted or not by checking if it's `None`. However, the above example does not work, because uploaded files are never `None`. Example output, even if no file is uploaded:

```bash
<starlette.datastructures.UploadFile object at ...> <starlette.datastructures.UploadFile object at ...>
<tempfile.SpooledTemporaryFile object at ...> <tempfile.SpooledTemporaryFile object at ...>
```

Thus, any subsequent processing of the files throws an exception (`char_count` or `audio_duration`).

## Solution to my problem
The solution came from [discord](https://discord.com/channels/772607987558252564/772631203689922562/1077653200863232031), and it's to check for the file `size` property.

## Suggestion
My code works fine now. I just wanted to point out the inconsistency in the documentation. If not implement the code that makes the file to be None, then perhaps, use `file.size == 0` in the documentation instead.

### Operating System

Linux

### FastAPI Version

0.92.0

### Python Version

3.11

### Additional Context

The endpoint works correctly if the files are uploaded as expected. The only problem is when checking for invalid files.

**Comment by jgould22** [comment]
Does it work correctly if you remove one of the values from your form in your upload get method?

When it says optional I am pretty sure it implies not sending the value in the form data.

Your form is submitting an empty value, but it is still part of the form. When they say optional they mean submitting form data without that field at all. 

ie the difference between 
```
curl localhost:<whatever port>/up
   -F text_file=@file.txt
``` 
and 
```
curl localhost:<whatever port>/up
   -F text_file=@file.txt
   -F audio_file=""
```

**Reply by regisin** [reply]
It works correctly both ways, so it's not that... sending empty field resolves to None by fastapi.
